rfcs/0001-reason-message-diff.md

```diff
@@ -2,52 +2,291 @@
 
-These one-word values are helpful for providing a basic level of context on why a new build has been created, but they don't provide enough information to inform users' decisions about what to do with the resulting build. They may be asking themselves: "Does this new build have some update that affects my app? Should I run my regression tests on it? Should I promote it to prod?" 
+These one-word values are helpful for providing a basic level of context on why a new build has been created, but they don't provide enough information to inform users' decisions about what to do with the resulting build. Based on user interviews, we know they are asking themselves: "Does this new build have some update that affects my app? Should I run my regression tests on it? Should I promote it to prod?" 
 
-Providing more information would reduce the friction in users taking action on new builds and increase customers' app security by making sure more patched builds make it into production faster. 
+When problems come up with an old build, they also ask themselves: "How do I know whats in this build that caused the problem? What makes this build different than the other ones that are working? Where should I start to troubleshoot that build?"
+
+Providing more information would reduce the friction in users taking action on new builds, resolve issues with older builds, and increase users' app security by making sure more patched builds make it into production faster. 
 
 ### Goals:
 
-Propose an additional attribute, “Reason-Message” be added to the build object, which can be printed in the context of the existing “Reason” attribute. Contextual information can be provided in the Reason-Message and a diff may also be printed as appropriate. This “Reason-Message” would be printed for these two commands:
+Propose that a diff or reason-message, as appropriate, be printed in the context of the existing “Reason” attribute. This would apply only to this command
 
 $ kp build status image-name
-$ kp image status image-name
 
-The format in which it would be printed:
+See mockups below. 
+
+### Prior Art:
+We have been thinking about different ways to surface the [Bill of Materials](https://github.com/buildpacks/pack/issues/378) for a build and how to show the delta between builds for a few months now. This is an evolution of that work. Although the `pack` cli renders the contents of a single BoM alone with the `inspect-image` command, I believe we don't should provide the BoM in this case, because this feature is intended to answer questions around WHY a build kicked off and whats different about the causes of the build, not the contents. It's important to address cause and contents separately because the dependencies involved are not 1:1 related.  A true diff of the contents would be better done using a different feature or standalone tool. 
+
+### Complexity/Risks:
+How does the user compare two non-adjacent builds? ex. builds 17 and 20
```

Comment:
Is this still a risk if you're de-scoping the build diff?

Reply:
i think the latter